frame/lending/src/lib.rs

```diff
@@ -1113,13 +1113,23 @@ pub mod pallet {
 				.ok_or_else(|| Error::<T>::MarketDoesNotExist.into())
 		}
 
+		/// Get TWAP from oracle. If history of prices is empty then return latest price.
 		fn get_price(
 			asset_id: <T as DeFiComposableConfig>::MayBeAssetId,
 			amount: T::Balance,
 		) -> Result<T::Balance, DispatchError> {
-			<T::Oracle as Oracle>::get_price(asset_id, amount)
-				.map(|p| p.price)
-				.map_err(|_| Error::<T>::AssetPriceNotFound.into())
+			const WINDOW: usize = 3;
+			let prices_length = <T::Oracle as Oracle>::depth_of_history(asset_id);
+			if prices_length == 0 {
```

Comment:
may be move it to <T::Oracle as Oracle>::get_price_weghted(asset_id, amount, weights)

Reply:
do you mean move all logic to `Oracle` side and remove `depth_of_history`? I would do it, but what should be result of the new method `Result<Option<Price>>` or `Result<Price>` with fallback to `get_price` inside?